packages/truffle-box/lib/utils/unbox.js

```diff
@@ -1,72 +1,63 @@
 const fs = require("fs-extra");
 const path = require("path");
 const ghdownload = require("github-download");
-const request = require("request");
+const rp = require("request-promise");
 const vcsurl = require("vcsurl");
 const parseURL = require("url").parse;
 const exec = require("child_process").exec;
 const inquirer = require("inquirer");
 
-function verifyURL(url) {
+async function verifyURL(url) {
   // Next let's see if the expected repository exists. If it doesn't, ghdownload
   // will fail spectacularly in a way we can't catch, so we have to do it ourselves.
-  return new Promise(function(accept, reject) {
-    var configURL = parseURL(
-      vcsurl(url)
-        .replace("github.com", "raw.githubusercontent.com")
-        .replace(/#.*/, "") + "/master/truffle-box.json"
-    );
+  const configURL = parseURL(
+    `${vcsurl(url)
+      .replace("github.com", "raw.githubusercontent.com")
+      .replace(/#.*/, "")}/master/truffle-box.json`
+  );
 
-    var options = {
-      method: "HEAD",
-      uri: "https://" + configURL.host + configURL.path
-    };
-    request(options, function(error, r) {
-      if (error) {
-        return reject(
-          new Error(
-            "Error making request to " +
-              options.uri +
-              ". Got error: " +
-              error.message +
-              ". Please check the format of the requested resource."
-          )
-        );
-      } else if (r.statusCode === 404) {
-        return reject(
-          new Error(
-            "Truffle Box at URL " +
-              url +
-              " doesn't exist. If you believe this is an error, please contact Truffle support."
-          )
+  const options = {
+    method: "HEAD",
+    uri: `https://${configURL.host}${configURL.path}`,
+    simple: false,
+    resolveWithFullResponse: true
+  };
+
+  await rp(options)
```

Comment:
Here I would either use `async/await` syntax OR `then` chains. I think it is confusing to use both and defeats the purpose of using `await` to begin with.

Reply:
Hmm, I'm getting wonky unhandled rejection behavior that appears related to the Bluebird promises powering `request-promise`. Going to give `request-promise-native` a shot and see if native ES6 promises resolve the issue. Otherwise, it appears the current `request` callback code is better suited to handling the current various error types.

Reply:
Are you handling the rejection cases?

Reply:
Even when I catch and throw after the rejection I still get an unhandled promise rejection response.

I _think_ I found a solution, but not really a fan of it. About to find out if `request-promise-native` makes a difference.

Reply:
Hmm, nm. In general the `request-promise` stuff doesn't seem to be integrating well and the current callback behavior appears to fit the error handling we output to users better. Going to see if I can just `promisify` `request`.

Reply:
Ahh, turns out there was an issue with the structuring of the sandbox method. Still can't seem to get a promise request to play nice w/o a mix of `await` and a `then` chain. Reverting the code to use a request callback for now.